### PR TITLE
Support Conversion from FHIR Stu3 to R4

### PR DESCRIPTION
**Pull request by sowu880**
## Description
A draft PR for internal review. 
Todo: Update version of converter engine after new release in Converter OSS.

Add new data type: "Fhir" to support Stu3 to R4.

## Testing
Added unit tests and E2E tests for the new feature.

## FHIR Team Checklist
- [x] **Update the title** of the PR to be succinct and less than 50 characters
- [x] **Add a milestone** to the PR for the sprint that it is merged (i.e. add S47)
- [x] Tag the PR with the type of update: **Bug**, **Dependencies**, **Enhancement**, or **New-Feature**
- [x] Tag the PR with Azure API for FHIR if this will release to the Azure API for FHIR managed service (CosmosDB or common code related to service)
- [x] Tag the PR with Azure Healthcare APIs if this will release to the Azure Healthcare APIs managed service (Sql server or common code related to service)
- [x] CI is green before merge
- Review [squash-merge requirements](https://github.com/microsoft/fhir-server/blob/master/SquashMergeRequirements.md)

### Semver Change ([docs](https://github.com/microsoft/fhir-server/blob/master/docs/Versioning.md))
Patch|Skip|Feature|Breaking (reason)

### CODE REVIEW

File: src/Microsoft.Health.Fhir.Core/Microsoft.Health.Fhir.Core.csproj [hunk 1]
```diff
@@ -49,7 +49,7 @@
     <PackageReference Include="Hl7.Fhir.Support.Poco" Version="$(Hl7FhirVersion)" />
     <PackageReference Include="Hl7.Fhir.Serialization" Version="$(Hl7FhirVersion)" />
     <PackageReference Include="Hl7.FhirPath" Version="$(Hl7FhirVersion)" />
-    <PackageReference Include="Microsoft.Health.Fhir.Liquid.Converter" Version="4.5.0.4" />
+    <PackageReference Include="Microsoft.Health.Fhir.Liquid.Converter" Version="4.6.0.31" />
```
Comment (sowu880) [inline src/Microsoft.Health.Fhir.Core/Microsoft.Health.Fhir.Core.csproj:52]: Todo: Update version when release.

**Comment by brendankowitz** [comment]
I think the last time I tested $convert-data it was returning `text` in the content-type header, wondering, is there ever a time when it doesn't return json?

**Comment by sowu880** [comment]
> I think the last time I tested $convert-data it was returning `text` in the content-type header, wondering, is there ever a time when it doesn't return json?

For now, it always returns FHIR R4 result in json format, but may support other conversion in the future.

File: src/Microsoft.Health.Fhir.Core/Microsoft - Backup.Health.Fhir.Core.csproj [hunk 2]
```diff
@@ -0,0 +1,81 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
```
Comment (brendankowitz) [inline src/Microsoft.Health.Fhir.Core/Microsoft - Backup.Health.Fhir.Core.csproj:1]: Should this file still be here?

Comment (sowu880) [reply src/Microsoft.Health.Fhir.Core/Microsoft - Backup.Health.Fhir.Core.csproj]: Thanks, removed this file.

**Review by brendankowitz — APPROVED** [review]

**Pull request merged**